### PR TITLE
pig test against real PNC

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
### Checklist:

* [X] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/bacon/wiki/Changelog) for your change if user-facing?
* [X] Have you added -unit- tests for your change?

### CODE REVIEW

**Comment by michalszynkiewicz** [comment]
the change also adds a posibility to overwrite config varialbes with system properties

**Comment by michalszynkiewicz** [comment]
It's much easier to write such tests against real PNC, and it also should make it much easier to reproduce the bugs, like the one we have for wrong build returned when the builds are skipped.

Next we could add mocks to be able to run it without the real systems (such duality, as I understand was @janinko's original idea)

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -93,7 +93,39 @@
...
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <excludedGroups>RealServiceOnly</excludedGroups>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>mock-tests</id>
+                    </execution>
+                </executions>
```
Comment (rnc) [inline integration-tests/pom.xml:105]: Do we need another execution here?

File: integration-tests/pom.xml [hunk 2]
```diff
@@ -93,7 +93,39 @@
...
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <excludedGroups>MockOnly</excludedGroups>
+                        </configuration>
+                        <executions>
+                            <execution>
+                                <id>real-tests</id>
+                            </execution>
```
Comment (rnc) [inline integration-tests/pom.xml:123]: Won't this run all the tests again?Don't we just want to run the functional test ones in this execution?

**Comment by rnc** [comment]
> It's much easier to write such tests against real PNC, and it also should make it much easier to reproduce the bugs, like the one we have for wrong build returned when the builds are skipped.
> 
> Next we could add mocks to be able to run it without the real systems (such duality, as I understand was @janinko's original idea)

Yes I agree, I often find most of the time is spent writing a test to reproduce the issue :-D 

Looks good, couple of comments about how tests are selected via maven.

**Comment by dwalluck** [comment]
We _could_ switch to using surefire for unit tests and failsafe for integration tests, then it's easier to just specify

```
<configuration>
    <groups>SomeGroup</groups>
</configuration>
```

one type of group (UnitTest, IntegrationTest, for example) in each plugin and no executions should be necessary.

File: integration-tests/pom.xml (hunk 2 above)
Comment (michalszynkiewicz) [reply integration-tests/pom.xml:123]: true, I haven't thought it through.

Looks like there needs to be another profile for mock only and all other tests could be executed in the "normal" execution. I'll update the PR

Comment (rnc) [reply integration-tests/pom.xml:123]: I would view all other tests, including mock as the default. Perhaps just run the real tests in the separate profile ?

**Comment by michalszynkiewicz** [comment]
I found a way of configuring surefire in a way that works ;)

The tests I marked as MockOnly are failing when tested with stage, if that's fixed it's enough to remove the tag from them to make them executed in both cases.

File: integration-tests/pom.xml (hunk 1 above)
Comment (michalszynkiewicz) [reply integration-tests/pom.xml]: no new execution now :)

File: integration-tests/pom.xml (hunk 2 above)
Comment (michalszynkiewicz) [reply integration-tests/pom.xml]: Now by default all tests by the real service tests are executed, and if switched to `functionalTests`, only the tests that work well with mocks are excluded.

**Comment by michalszynkiewicz** [comment]
I'm merging it, it doesn't impact "normal usage". Happy to refactor later if required

**Pull request merged**